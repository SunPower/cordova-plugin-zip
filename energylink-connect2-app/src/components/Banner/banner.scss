.development-banner {
  position: fixed;
  background-color: indianred;
  top: 17px;
  right: -27px;
  transform: rotateZ(45deg);
  width: 100px;
  text-align: center;
  color: white;
  z-index: 99999;
}
