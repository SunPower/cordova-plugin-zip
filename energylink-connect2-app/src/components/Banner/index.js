import React from 'react'
import './banner.scss'
import { either, isDebug } from 'shared/utils'

const Banner = ({ flavor = '' }) =>
  either(
    isDebug,
    <div className="is-capitalized development-banner">
      {flavor.split('-').pop()}
    </div>
  )

export default Banner
