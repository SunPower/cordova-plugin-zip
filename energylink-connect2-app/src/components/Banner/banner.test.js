import { shallow } from 'enzyme'
import React from 'react'
import Banner from './index'

describe('Ribbon should be', function() {
  it('hidden in Prod and Training flavors', function() {
    const component = shallow(<Banner flavor="cm2-prod" />)
    expect(component).toMatchSnapshot()
    const training = shallow(<Banner flavor="cm2-training" />)
    expect(training).toMatchSnapshot()
  })
  it('displayed in Test and UAT flavors', function() {
    const component = shallow(<Banner flavor="cm2-test" />)
    expect(component).toMatchSnapshot()
    const uat = shallow(<Banner flavor="cm2-uat" />)
    expect(uat).toMatchSnapshot()
  })
})
