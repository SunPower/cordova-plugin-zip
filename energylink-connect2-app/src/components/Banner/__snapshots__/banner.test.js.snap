// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ribbon should be displayed in Test and UAT flavors 1`] = `""`;

exports[`Ribbon should be displayed in Test and UAT flavors 2`] = `""`;

exports[`Ribbon should be hidden in Prod and Training flavors 1`] = `""`;

exports[`Ribbon should be hidden in Prod and Training flavors 2`] = `""`;
